### PR TITLE
Add optional totals to TableComponent

### PR DESCRIPTION
**Pull request by arzezak**
We need to display total rows in order to match existing designs.

Screenshot 2023-09-07 at 11 43 52 AM

### CODE REVIEW

File: spec/components/anchor/table_component_spec.rb [hunk 1]
```diff
@@ -149,4 +149,67 @@ def [](key)
...
+    context "with no argument" do
+      it "renders no total" do
+        component = described_class.new(data: [])
+
+        render_inline(component) do |table|
+          table.with_column(header: :id, value: :id)
+          table.with_column(header: :name, value: :name)
+        end
+
+        expect(page).not_to have_text "Total: 0"
```
Comment (tute) [inline spec/components/anchor/table_component_spec.rb:163]: I'd remove the zero so we know it doesn't appear regardless of spec setup (no need to double check):
```suggestion
        expect(page).not_to have_text "Total:"
```

File: spec/components/anchor/table_component_spec.rb [hunk 2]
```diff
@@ -149,4 +149,67 @@ def [](key)
...
+      end
+    end
+
+    context "single record" do
```
Comment (tute) [inline spec/components/anchor/table_component_spec.rb:167]: Do we need to distinguish this context?

**Review by tute — APPROVED** [review]

File: app/components/anchor/table_component.rb [hunk 3]
```diff
@@ -3,20 +3,22 @@ class TableComponent < Component
...
+    attr_reader :data, :sortable, :caption, :total
 
     def initialize(
       data:,
       caption: nil,
       empty_message: nil,
       sort_url: nil,
-      sortable: false
+      sortable: false,
+      total: false
```
Comment (Simonpedro) [inline app/components/anchor/table_component.rb:14]: Totally subjective
```suggestion
      show_total: false
```

**Review by Simonpedro — APPROVED** [review]

File: app/components/anchor/table_component.rb (hunk 3 above)
Comment (arzezak) [reply app/components/anchor/table_component.rb:14]: @tute, we need you to break even.

Comment (tute) [reply app/components/anchor/table_component.rb:14]: I like `show_total` better too!

File: app/components/anchor/table_component.html.erb [hunk 4]
```diff
@@ -41,7 +41,13 @@
     end %>
   </table>
 
-  <% if data.empty?  %>
+  <% if total %>
+    <div class="mt-3 text-base font-semibold">
+      <%= "Total: #{data.size}" %>
```
Comment (dgalarza) [inline app/components/anchor/table_component.html.erb:46]: We should likely put this in i18n.

Comment (arzezak) [reply app/components/anchor/table_component.html.erb]: 👍 Makes sense.

**Review by dgalarza — APPROVED** [review]

**Pull request merged**